bite/BiteCommittedBlockSerializer.cpp

```diff
@@ -0,0 +1,230 @@
+
+
+    const skale_fb::CommittedBlock *fbBlock = nullptr;
+
+    VERIFY_AND_PARSE_FLATBUFFER_FROM_VECTOR(*_serializedBlock, CommittedBlock, fbBlock);
+
+    // Extract block header data
+    auto fbHeaderVec = fbBlock->block_header();
+    CHECK_STATE(fbHeaderVec);
+    size_t headerSize = fbHeaderVec->size();
+    CHECK_STATE(fbHeaderVec->data())
+    std::string_view headerView(reinterpret_cast<const char *>(fbHeaderVec->data()), headerSize);
+
+    ptr<CommittedBlockHeader> blockHeader;
+
+    try {
+        blockHeader = CommittedBlock::parseBlockHeader(headerView);
+        CHECK_STATE(blockHeader);
+    } catch (...) {
+        throw_with_nested(ParsingException(
+            "Could not parse committed block header: \n" + string(headerView), __CLASS_NAME__));
+    }
+
+
+    auto fbTransactions = fbBlock->transactions();
+    CHECK_STATE(fbTransactions);
+
+    auto transactions = make_shared<vector<ptr<Transaction> > >();
+    transactions->reserve(fbTransactions->size());
+
+    for (const auto *tx: *fbTransactions) {
+        CHECK_STATE(tx)
+        auto rawData = tx->data()->data();
+        CHECK_STATE(rawData);
+        auto txData = make_shared<vector<uint8_t> >(rawData,
+                                                    rawData + tx->data()->size());
+        auto txObj = make_shared<Transaction>(txData, false); // hypothetical
+        transactions->push_back(txObj);
+    }
+
+    auto transactionList = std::make_shared<TransactionList>(transactions);
+
+    // now deserialize AES keys
+
+    ptr<DecryptedAESKeyList> decryptedAesKeyList = make_shared<DecryptedAESKeyList>();
+
+    auto fbAesKeys = fbBlock->aes_keys();
+
+    CHECK_STATE(fbAesKeys)
+
+    if (fbAesKeys) {
+        for (const auto *aesKey: *fbAesKeys) {
+            CHECK_STATE(aesKey->data() && aesKey->data()->size() == BITE_AES_KEY_LEN);
+            std::array<uint8_t, BITE_AES_KEY_LEN> rawKey;
+            std::memcpy(rawKey.data(), aesKey->data()->data(), BITE_AES_KEY_LEN);
+
+            DecryptedAESKey key(rawKey);
+            decryptedAesKeyList->addKey(aesKey->transaction_index(), key);
+        }
+    }
+
+    auto decryptedTransactionDataFields = _biteManager->verifyAndDecryptTransactionList(*transactionList,
+        *decryptedAesKeyList);
+    ptr<CommittedBlock> block = nullptr;
+
+    try {
+        block = CommittedBlock::make(blockHeader->getSchainID(), blockHeader->getProposerNodeId(),
+                                     blockHeader->getBlockID(), blockHeader->getProposerIndex(), transactionList,
+                                     blockHeader->getStateRoot(), blockHeader->getTimeStamp(),
+                                     blockHeader->getTimeStampMs(),
+                                     blockHeader->getSignature(), blockHeader->getThresholdSig(),
+                                     blockHeader->getDaSig()
+#ifdef BITE
+                                     , decryptedAesKeyList, decryptedTransactionDataFields
```

Comment:
this file is included to the project only if BITE is defined. I think we can remove ifdef here